docs/rules/no-undefined.md

```diff
@@ -14,24 +14,15 @@ function doSomething(data) {
 }
 ```
 
-This represents a problem for `undefined` that doesn't exist for `null`, which is a keyword and primitive value that can neither be overwritten nor shadowed.
+Because `undefined` can be overwritten or shadowed, reading `undefined` can give an unexpected value. (This is not the case for `null`, which is a keyword that always produces the same value.) To guard against this, you can use the [no-global-assign](no-global-assign.md) and [no-shadow-restricted-names](no-shadow-restricted-names.md) rules.
 
-All uninitialized variables automatically get the value of `undefined`:
+Alternatively, you can avoid all uses of `undefined`, which is what some style guides recommend. Those style guides then recommend:
```

Comment:
In the context of this rule, it doesn't make sense to say an `undefined`-free coding style is "alternative". In this rule, that would be the primary use case. So I would lead this paragraph by saying that this rule allows you to avoid all uses of `undefined`. Then, below the style guide recommendations, I would suggest moving the excerpt I called out above (starting with "To guard against this") and make that paragraph the new "alternative" paragraph.

The alternative to using this rule (for flagging *all* uses of `undefined`) is to enable those two rules, which will flag *writes* to undefined but allow reads.

Hope this makes sense and doesn't sound arbitrary. Let me know if I can clarify anything!